alembic/versions/feeaa70ead67_social_authentication_table.py

```diff
@@ -1,23 +1,109 @@
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
 """Social authentication table
 
 Revision ID: feeaa70ead67
 Revises: a52d2fe99d4f
 Create Date: 2017-07-07 14:23:30.605238
 """
+import uuid
 
 from alembic import op
 from social_sqlalchemy import storage
+from sqlalchemy.dialects.postgresql import UUID
+from sqlalchemy.types import TypeDecorator, CHAR
 import sqlalchemy as sa
 
-from anitya.lib.model import GUID
-
 
 # revision identifiers, used by Alembic.
 revision = 'feeaa70ead67'
 down_revision = 'a52d2fe99d4f'
 
 
+class GUID(TypeDecorator):
+    """
+    Platform-independent GUID type.
+
+    If PostgreSQL is being used, use its native UUID type, otherwise use a CHAR(32) type.
+    """
+    impl = CHAR
+
+    def load_dialect_impl(self, dialect):
+        """
+        PostgreSQL has a native UUID type, so use it if we're using PostgreSQL.
+
+        Args:
+            dialect (sqlalchemy.engine.interfaces.Dialect): The dialect in use.
+
+        Returns:
+            sqlalchemy.types.TypeEngine: Either a PostgreSQL UUID or a CHAR(32) on other
+                dialects.
+        """
+        if dialect.name == 'postgresql':
+            return dialect.type_descriptor(UUID())
+        else:
+            return dialect.type_descriptor(CHAR(32))
+
+    def process_bind_param(self, value, dialect):
+        """
+        Process the value being bound.
+
+        If PostgreSQL is in use, just use the string representation of the UUID.
+        Otherwise, use the integer as a hex-encoded string.
+
+        Args:
+            value (object): The value that's being bound to the object.
+            dialect (sqlalchemy.engine.interfaces.Dialect): The dialect in use.
+
+        Returns:
+            str: The value of the UUID as a string.
+        """
+        if value is None:
+            return value
+        elif dialect.name == 'postgresql':
+            return str(value)
+        else:
+            if not isinstance(value, uuid.UUID):
+                return "%.32x" % uuid.UUID(value).int
+            else:
+                # hexstring
+                return "%.32x" % value.int
+
+    def process_result_value(self, value, dialect):
+        """
+        Casts the UUID value to the native Python type.
+
+        Args:
+            value (object): The database value.
+            dialect (sqlalchemy.engine.interfaces.Dialect): The dialect in use.
+
+        Returns:
+            uuid.UUID: The value as a Python :class:`uuid.UUID`.
+        """
+        if value is None:
+            return value
+        else:
+            return uuid.UUID(value)
```

Comment:
It might be a nice compromise to have only the first migration that needs this copy of ```GUID``` to copy it from the models, and have the other migrations that use the exact same code import it from the migration. This way you avoid the "what if GUID changes in some way that breaks the migration" problem, but you also don't have to have n copies of it.